src/Symfony/Component/Mime/WrappedMessage.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime;
+
+/**
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * @experimental in 4.3
+ */
+class WrappedMessage extends RawMessage
+{
+    private $originalMessage;
+
+    /**
+     * @param iterable|string $message
+     */
+    public function __construct(Message $originalMessage, $message)
+    {
+        $this->originalMessage = $originalMessage;
+        $this->message = $message;
+    }
+
+    public function getOriginalMessage(): Message
+    {
+        return $this->originalMessage;
+    }
+
+    /**
+     * @param string|iterable $wrappedMessage
+     *
+     * @return $this
+     */
+    public function setMessage($wrappedMessage)
+    {
+        $this->message = $wrappedMessage;
+
+        return $this;
+    }
+
+    /**
+     * @return string|iterable
+     */
+    public function getMessage()
+    {
+        return $this->message;
+    }
+
+    /**
+     * @internal
+     */
+    public function __serialize(): array
+    {
+        $this->toString();
```

Comment:
Why is it needed?

Reply:
The SMIME signer/encryptor uses a generator that cannot be serialized. In fact, `tmpfile()` removes the file after a successful shutdown. Keeping theses files for a longer period would mean extensive growth of temp files.

Alternatively we can store the S/MIME as string, but this will require more memory.